src/test/java/org/kiwiproject/registry/consul/server/ConsulRegistryServiceIntegrationTest.java

```diff
@@ -0,0 +1,199 @@
+package org.kiwiproject.registry.consul.server;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.common.net.HostAndPort;
+import com.orbitz.consul.Consul;
+import com.orbitz.consul.model.agent.ImmutableRegistration;
+import com.orbitz.consul.model.catalog.CatalogService;
+import com.pszymczyk.consul.junit.ConsulExtension;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.kiwiproject.base.KiwiEnvironment;
+import org.kiwiproject.registry.consul.config.ConsulRegistrationConfig;
+import org.kiwiproject.registry.exception.RegistrationException;
+import org.kiwiproject.registry.model.ServiceInstance;
+import org.kiwiproject.registry.util.ServiceInfoHelper;
+
+import java.time.Instant;
+
+@DisplayName("ConsulRegistryService")
+@ExtendWith(SoftAssertionsExtension.class)
+class ConsulRegistryServiceIntegrationTest {
+
+    // NOTE: Even though this extension uses an AfterAllCallback, it can NOT be static as running all of the tests fail. I'm not sure if this is
+    //       something with the extension or with the Nested test classes
+    @RegisterExtension
+    public ConsulExtension CONSUL = new ConsulExtension();
```

Comment:
This should work as package-private.

I am _not_ sure why it cannot be static. Should we add a TODO here or add an "investigation" issue to determine why?

Reply:
Made it package-private.  I commented about why it is not working.  I think it is based on the way the `ConsulExtension` was written.  It works if I run a single test, it also works if I run a single `@Nested` class, but not the whole class itself.  The extension is doing things with `AfterAll` and `AfterEach`.